src/common/decorators/subscription.decorator.ts

```diff
@@ -0,0 +1,49 @@
+import { IncomingHttpHeaders } from 'http';
+import {
+  ReturnTypeFunc,
+  Subscription,
+  SubscriptionOptions,
+} from '@nestjs/graphql';
+import { GraphQLResolveInfo } from 'graphql';
+import { AgentInfo } from '@src/core';
+
+export interface SubscriptionResolveContext {
```

Comment:
one interface / type per file is standard across the server, please split out into a separate file